### ISSUE TITLE
Rule Request: split `identifier_name` -> `short_identifier` + `long_identifier` + `camel_identifier_name`

### ISSUE THREAD

**Issue by mleonhard** [realm/SwiftLint#3981]
### New Issue Checklist

- [X ] Updated SwiftLint to the latest version
- [X ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

I wish to use the identifier `n` for iterator index.  Some short functions use `s` for a `String` parameter and `x` for an `Int` parameter.

I occasionally need to write a very long class name.  This usually happens when sub-classing a library class.

I propose that we split the [identifier_name](https://realm.github.io/SwiftLint/identifier_name.html) rule into three rules:
- `short_identifier` requiring identifiers to be 3 characters or longer
- `long_identifier` requiring identifiers to be 40 characters or shorter
- `camel_identifier_name` requiring identifiers to be lower-camel-case

The new rules should be opt-in.  Let's add a comment to the `identifier_name` docs.

**Comment by mleonhard** [comment]
I learned that I can configure `identifier_name` rule to allow single-character names and long names:
```yaml
# .swiftlint.yml
identifier_name:
  min_length:
    warning: 1
    error: 1
  max_length:
    warning: 100
    error: 100
```

Closing.
